rust/src/meas/randomized_response/mod.tex

```diff
@@ -0,0 +1,250 @@
+        
+    %    Let $p$ represent \emph{prob}; then the expected value of the response is $p \cdot \text{\emph{arg}} + (1 - p)(1 - \text{\emph{arg}})$. \vicki{idk why i did this but may be relevant} 
+    %\end{enumerate}
+    %\item \textbf{(Appropriate output domain.)} In the case of \texttt{binary\_randomized\_response}, this corresponds to showing that for every element $v$ in \texttt{AllDomain(bool)}, the element \texttt{function(v)} belongs to \texttt{AllDomain(bool)}. The type signature of \texttt{function} as defined in line \ref{line:fn} automatically enforces that \texttt{function(v)} has type \texttt{bool}. Since the Rust code successfully compiles, by the type signature the appropriate output domain property must hold. Otherwise, the code will raise an exception for incorrect input type. 
+    
+    %Another way to see this is that \texttt{function(v)} returns either \emph{arg} or \emph{lie}. From the function signature, \emph{arg} is of type \texttt{bool}. As shown above, \texttt{sample\_bernoulli} can either return 0 or 1, so it is therefore also restricted to boolean values. Hence, \emph{lie} must also be boolean. 
+    
+    %Therefore, \texttt{function(v)} is guaranteed to be in \texttt{output\_domain} in all cases.
+    
+    % need to check for overflow? 
+
+    \item \textbf{(Domain-metric compatibility.)} For \texttt{binary\_randomized\_response}, this corresponds to showing \texttt{AllDomain(bool)} is compatible with \texttt{DiscreteMetric}. This follows directly from the definition of \texttt{DiscreteMetric}, as stated in the \href{https://github.com/opendp/whitepapers/blob/pseudocode-defns/pseudocode-defns/pseudocode_defns.pdf}{``List of definitions used in the pseudocode"}.
+    
+    \item \textbf{(Privacy guarantee.)} 
+    % We assume proof of correctness of the \texttt{bernoulli\_sampler} as shown in the document ??? (to do). 
+    
+    \begin{tcolorbox}
+\begin{note}[Proof relies on correctness of Bernoulli sampler]
+The following proof makes use of the following lemma that asserts the correctness of the Bernoulli sampler function.
+    \begin{lemma}
+    \texttt{sample\_bernoulli(prob, constant\_time)}, the Bernoulli sampler function used in \texttt{make\_randomized\_response\_bool}, returns \texttt{true} with probability (\texttt{prob}) and returns  \texttt{false} with probability (1 - \texttt{prob}).
```

Comment:
Can sample_bernoulli fail?  (If so, the lemma would need to incorporate this, and the use of the lemma would have to take this into account.) Also I'd move the reliance on this lemma to the Warning section, since we don't yet have a proof of the lemma.

Reply:
The function returns errors that are not data-dependent. The primary (and unlikely) cause of this is openssl failing due to lack of system entropy.

Reply:
Should we still account for this in the proof? Or is it ok to just mention as a caveat that `sample_bernoulli` could fail due to OpenSSL failing because of lack of system entropy?

Reply:
This was something Salil brought up in the meeting on Thursday. The proof should include failure as a possible output of the mechanism, and first show that releasing a failure state is independent from the dataset. Then analyze the true/false releases as a conditional probability, conditioned on the assumption that failure is not released.